From bc9dbfb90f23b79833c7a32d8992fa71b38753d1 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Wed, 14 Mar 2012 20:11:18 +0200
Subject: [PATCH 064/370] add function names to state labels

This change makes the state diagram more informative.
Also describe ->sd_in() of S_TERMINAL and S_FAILURE.

Reported-by: Nikita Danilov <nikita_danilov@xyratex.com>
Reviewed-on: http://reviewboard.clusterstor.com/r/714/
---
 conf/confc.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/conf/confc.c b/conf/confc.c
index 374e1de..e6e4cd0 100644
--- a/conf/confc.c
+++ b/conf/confc.c
@@ -180,8 +180,14 @@
  *     node [fontsize=11];
  *     edge [fontsize=11];
  *     S_INITIAL  [style=filled, fillcolor=lightgrey];
- *     S_TERMINAL [style=filled, fillcolor=lightgrey];
- *     S_FAILURE  [style=filled, fillcolor=lightgrey];
+ *     S_TERMINAL [style=filled, fillcolor=lightgrey,
+ *                 label="{ completion_announce() }\nS_TERMINAL"];
+ *     S_FAILURE  [style=filled, fillcolor=lightgrey,
+ *                 label="{ completion_announce() }\nS_TERMINAL"];
+ *
+ *     S_CHECK [label="{ check_st_in() }\nS_CHECK"];
+ *     S_WAIT_REPLY [label="{ wait_reply_st_in() }\nS_WAIT_REPLY"];
+ *     S_GROW_CACHE [label="{ grow_cache_st_in() }\nS_GROW_CACHE"];
  *
  *     S_INITIAL -> S_CHECK;
  *     S_CHECK -> S_TERMINAL [label="all \"needed\" obj-s\nare cached"];
@@ -223,6 +229,10 @@
  *   grow_cache_st_in() moves the state machine to S_CHECK state upon
  *   success and to S_FAILURE in case of error.
  *
+ * - When S_TERMINAL or S_FAILURE state is entered,
+ *   completion_announce() is invoked.  This function broadcasts
+ *   c2_confc_ctx::fc_complete channel.
+ *
  * @subsection confc-dld-lspec-thread Threading and Concurrency Model
  *
  * When a confc instance is started (c2_confc_init(), it gets
-- 
1.8.3.2

